charts/dvpe-cluster-issuer/templates/cluster-issuer.yaml

```diff
@@ -1,30 +1,28 @@
-apiVersion: cert-manager.io/v1
+{{- $clcm := .Values.clcm }}
+{{- range .Values.clusterIssuers }}
+---
+apiVersion: controller-manager.dvpe-cloud.github.io/v1alpha1
 kind: ClusterIssuer
 metadata:
-  name: {{ .Values.issuer.metadata.name }}
-  namespace: {{ .Values.issuer.metadata.namespace }}
+  name: {{ .name }}
 spec:
-  acme:
-    {{- if .Values.issuer.spec.acme.server.useProd }}
-    server: {{ .Values.issuer.spec.acme.server.prod }}
-    {{- else }}
-    server: {{ .Values.issuer.spec.acme.server.staging }}
-    {{- end}}
-    email: {{ .Values.issuer.spec.acme.email }}
-    privateKeySecretRef:
-      {{- if .Values.issuer.spec.acme.server.useProd }}
-      name: issuer-prod-secret
-      {{- else }}
-      name: issuer-staging-secret
-      {{- end}}
-    solvers:
-        - selector:
-            dnsZones:
-              {{- range .Values.issuer.spec.acme.solvers.selector.dnsZones }}
-              - {{ . | quote }}
-              {{- end }}
-          dns01:
-            route53:
-              region: {{ .Values.issuer.spec.acme.solvers.dns01.route53.region }}
-              hostedZoneID: {{ .Values.issuer.spec.acme.solvers.dns01.route53.hostedZoneID }}
-              role: {{ .Values.issuer.spec.acme.solvers.dns01.route53.role }}
+  certificateDetails:
+    caInstance: {{ .caInstance }}
+    ciID: {{ .ciID }}
+    ciType: {{ .ciType }}
+    certificateDataCAType: {{ .certificateDateCAType }}
+    client: {{ .ciClient }}
+    contactEmail: {{ .ciContactEmail }}
+    requester: {{ .ciRequester }}
+  clcm:
```

Comment:
aren't those parameters mandatory? instead of using ifs it might make sense to use a json schema in order to validate those params

Reply:
also, maybe it makes sense to have two specific sections for public and private issuers instead of the more generic loop. this would facilitate validation